http4s/server-circe/src/main/scala/endpoints/http4s/server/circe/JsonEntitiesFromCodec.scala

```diff
@@ -0,0 +1,44 @@
+package endpoints.http4s.server.circe
+
+import cats.implicits._
+import endpoints.algebra.Documentation
+import endpoints.http4s.server.Endpoints
+import io.circe.Json
+import io.circe.parser._
+import org.http4s
+import org.http4s.circe._
+import org.http4s.{DecodeFailure, InvalidMessageBodyFailure}
+
+trait JsonEntitiesFromCodec[F[_]]
+    extends Endpoints[F]
+    with endpoints.algebra.circe.JsonEntitiesFromCodec {
+
+  def jsonRequest[A](docs: Documentation = None)(
+      implicit codec: JsonRequest[A]): RequestEntity[A] =
+    req => {
+      def transform(t: Either[DecodeFailure, Json]): Either[DecodeFailure, A] =
+        for {
+          str <- t
```

Comment:
This will not compile with 2.11. The workaround consists in adding a `.right` call (`str <- t.right`).